polaris/hub/client.py

```diff
@@ -610,28 +610,24 @@ def upload_dataset(
         # Step 3: Upload any associated Zarr archive
         if dataset.zarr_root is not None:
             with tmp_attribute_change(self.settings, "default_timeout", timeout):
-                # Copy the Zarr archive to the hub
-                dest = self.open_zarr_file(
-                    owner=dataset.owner,
-                    name=dataset.name,
-                    path=dataset_json["zarrRootPath"],
-                    mode="w",
-                    as_consolidated=False,
+                polaris_fs = PolarisFileSystem(
+                    polaris_client=self,
+                    dataset_owner=dataset.owner,
+                    dataset_name=dataset.name,
                 )
+                dest = zarr.storage.FSStore(url=dataset_json["zarrRootPath"], fs=polaris_fs, mode="w")
 
                 # Locally consolidate Zarr archive metadata. Future updates on handling consolidated
                 # metadata based on Zarr developers' recommendations can be tracked at:
                 # https://github.com/zarr-developers/zarr-python/issues/1731
                 zarr.consolidate_metadata(dataset.zarr_root.store.store)
-                zmetadata_content = dataset.zarr_root.store.store[".zmetadata"]
-                dest.store[".zmetadata"] = zmetadata_content
```

Comment:
Why was this deleted?? If we can delete it here, can we delete it in `to_json` too?

Reply:
The changes resulted in `zarr.errors.CopyError: key '.zmetadata' exists in destination` so I had to remove it. Should also be good to remove it from `to_json`.

Reply:
Interesting! So `copy_store` solves https://github.com/zarr-developers/zarr-python/issues/1731?